### PR TITLE
Ld dt firebase setup

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.
      AC:
      *  [`firebase`](https://www.npmjs.com/package/firebase) has been added as a project dependency
      * Making a change in the Firestore database will update the data in the app
      * Making a change in the app will update the data in Firestore
      Notes:
      * Don't do too much here. In the past, pairs have built the whole add item form in the first week. The best approach is just to make a throw away view with a `<button/>` and a `<div/>`. Click the `<button/>` to send some data to the database. Have the data show up in the `<div/>`. Simple!
      * If you need additional direction on how to use Firebase, you can read through the [Quickstart Guide](https://firebase.google.com/docs/firestore/quickstart) for understanding how to [add items to Firestore](https://firebase.google.com/docs/firestore/quickstart#add_data), as well as the Snapshot Guide for [creating a snapshot subscription](https://firebase.google.com/docs/firestore/query-data/listen#listen_to_multiple_documents_in_a_collection) to read and update data in real-time.
      ---
      Quick Tips:
      <details>
        <summary>How to send data to Firestore</summary>
        ```javascript
      import { collection, addDoc } from 'firebase/firestore'
      try {
          const docRef = await addDoc(collection(db, COLLECTION_NAME), {
            property: 'value'
            // ...data
          })
          // Success!
          console.log(docRef.id)
      } catch (e) {
          // There was an error sending data to Firestore
          console.error(e)
      }
        ```
      </details>
      <details>
        <summary>How to listen to data from Firestore</summary>
        ```javascript
      import { collection, onSnapshot } from 'firebase/firestore'
      const [docs, setDocs] = useState([])
      useEffect(() => {
          const unsubscribe = onSnapshot(collection(db, COLLECTION_NAME), (snapshot) => {
            // Firestore requires that you loop through the snapshot to access the docs,
            // instead of just setting the snapshot as the value of the state
            const snapshotDocs = []
            snapshot.forEach(doc => snapshotDocs.push(doc))
            setDocs(snapshotDocs)
          })
          return () => {
            // Used to remove the snapshot listener when the component is unmounted
            unsubscribe();
          }
      }, [])
        ```
      </details>

### PR DESCRIPTION
**Pull request by djtaylor8**
## Type of Changes
|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |
## Acceptance Criteria
 [X] Firebase and firestore added to dependencies
 [X] App listens to database changes and displays changes
 [X] Make a change in the app and the data shows up in Firestore
## Related Issue
[Closes #1](https://github.com/the-collab-lab/tcl-36-smart-shopping-list/issues/1)
### Description
This PR adds two new states and functionalities: one for adding items to firestore and another for displaying the items.
### Display items from database
In the App.js we've added a button to handle submitting new items, an input for the user to type data, a button to submit the data from the input field, and a div to display new data from the firebase collection.
You can read more about the firebase setup [here](https://firebase.google.com/docs/firestore/quickstart)
# Testing Steps
- From your terminal, pull down this branch with `git pull origin ld-dt-firebase-setup` and check that branch out with `git checkout ld-dt-firebase-setup`
- Then `yarn` or `npm` install the newly added dependencies locally and `yarn start` or `npm start` to launch the app.
- Fill out the input box with a new item. Make sure it adds it to the div below the input and button and make sure that item appears in the items collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-36-smart-shopping-list/firestore/data/~2Fitems~2Fy5SteQP9oNovRl6WJhhQ).
 
Assignees
@djtaylor8, @lindseyindev

### CODE REVIEW

**Comment by alenamedved** [comment]
Good job, I followed 'Testing steps' and was able to add new items to db.

Just have a few comments:
1. We probably want to clean an input after we clicked the button.
2. Maybe we need another name for the state (not text, but item) as well for 'placeholder' and 'name' so it  describes better the information represented. 
3. Also I don't see how the 'App listens to database changes and displays changes' is implemented. The app doesn't read from the db yet. Are you still working on it? Descriptions says it has two new states but I can see only one.

**Comment by djtaylor8** [comment]
> Good job, I followed 'Testing steps' and was able to add new items to db.
> 
> Just have a few comments:
> 
> 1. We probably want to clean an input after we clicked the button.
> 2. Maybe we need another name for the state (not text, but item) as well for 'placeholder' and 'name' so it  describes better the information represented.
> 3. Also I don't see how the 'App listens to database changes and displays changes' is implemented. The app doesn't read from the db yet. Are you still working on it? Descriptions says it has two new states but I can see only one.

**Comment by djtaylor8** [comment]
Thank you, Alena! Good catch on clearing the input after it gets submitted - we will fix that and update 'text' to 'item'. To answer your last question - the app is listening to the database. The useEffect function in App.js is what connects the client side to Firestore (database), so every time an item is submitted from the form, it is saved and the page is updated with that new information. Unless we are missing something, I believe this should be working.

**Comment by alenamedved** [comment]
> Thank you, Alena! Good catch on clearing the input after it gets submitted - we will fix that and update 'text' to 'item'. To answer your last question - the app is listening to the database. The useEffect function in App.js is what connects the client side to Firestore (database), so every time an item is submitted from the form, it is saved and the page is updated with that new information. Unless we are missing something, I believe this should be working.

Oh, ok. Look like I we have a technical issue.  I don't see a useEffect function in App.js. Is it only me or you didn't push all your commits?

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
+import React, { useState } from 'react';
+import { collection, addDoc } from 'firebase/firestore';
+import { db } from './lib/firebase';
 import './App.css';
 
 function App() {
+  const [text, setText] = useState('');
+
+  //handle submit for the form
+  const handleClick = async (e) => {
```
Comment (alenamedved) [inline src/App.js]: how about if we be more specific and call it handleAddItem function or something like that

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+        text,
+      });
+      console.log('Document written with ID: ', docRef.id);
+    } catch (e) {
+      console.error('Error adding document: ', e);
+    }
+  };
+
+  //updating state for the input box
+  function handleChange(e) {
```
Comment (alenamedved) [inline src/App.js:40]: and something like handleItemChange here

**Comment by djtaylor8** [comment]
Sorry about that! The useEffect function should be included now. Let us know if you don't see it.

**Comment by alenamedved** [comment]
> Sorry about that! The useEffect function should be included now. Let us know if you don't see it.

Yeap, I see it now! Thanks. 
Do I need to remove my previous comments or we can keep it?

**Comment by djtaylor8** [comment]
You can keep those. We will get those things updated.

**Review by alenamedved — APPROVED** [review]
Really nice. 
And I love the comments and how quickly you updated the code to clean the input after submission.

File: src/App.js [hunk 3]
```diff
@@ -1,24 +1,67 @@
...
+            name="item"
+            value={text}
+            onChange={(e) => handleChange(e)}
+          ></input>
+          <button type="submit" onClick={(e) => handleClick(e)}>
+            Click this button to send data!
+          </button>
+        </form>
+        {docs.map((doc) => (
+          <p>{doc.text}</p>
```
Comment (alenamedved) [inline src/App.js]: <p> element missing a "key" prop, you should see a warning in your console

File: src/App.js [hunk 4]
```diff
@@ -1,24 +1,66 @@
...
+      unsubscribe();
+    };
+  }, []);
+
+  //handle submit for the form
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    //create new doc reference for text input and save to firestore database
+    try {
```
Comment (laurenyz) [inline src/App.js:28]: Great use of try and catch!

**Comment by rudidev08** [comment]
I'm big fan of how detailed testing steps are. While in theory all that information can be figured out (git branch, how to build, etc), there's big accessibility win that one can just copy & paste and test without much context switching.

If you were writing this app from scratch, more generic build & test steps would make a decent README.md file Speaking of, as project goes on, see if README.md would benefit with updates like that (optional and only if it makes sense & there's time! use it as your reminder pad how to build and test, ideally in format you can copy and paste)

**Review by rudidev08 — APPROVED** [review]

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,66 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <div className="temp">
```
Comment (laurenyz) [inline src/App.js]: Is `temp` a placeholder for future styling? I didn't see it in the css files.

**Review by laurenyz — APPROVED** [review]
Looks great! Very Clean! The comments are very helpful in understanding the functionality at a quick glance. I think some of the comments could be removed if some of the variables had names more closely related to their function. 
Ex. - instead of:
```
//use text and setText for form inputs
const [text, setText] = useState('');
//docs and setDocs will handle fetching items from firestore and displaying them in our div below
const [docs, setDocs] = useState([]);
 ``` 
 you could do...
 ```
 const [itemText, setItemText] =  useState('')
 const [items, setItems] = useState([]);
```

I generally try to describe the functionality via variable/function names first, and then add comments if further explanation is necessary so that they aren't distracting from the code. It also helps later on as you start adding other pieces of state.
 
Great PR! I'm excited to hear what you think of Firebase!

File: src/App.js (hunk 5 above)
Comment (djtaylor8) [reply src/App.js]: Oops! That's leftover from the boilerplate info when you first open a React app. We can remove that...

**Pull request merged**